L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -48,6 +63,9 @@ initialize(TTree& tree, const edm::ParameterSet& conf, edm::ConsumesCollector&&
 
     ee_token_ = collector.consumes<HGCEEDigiCollection>(conf.getParameter<edm::InputTag>("HGCDigisEE")); 
     fh_token_ = collector.consumes<HGCHEDigiCollection>(conf.getParameter<edm::InputTag>("HGCDigisFH"));
+    SimHits_inputee_ = collector.consumes<edm::PCaloHitContainer>(conf.getParameter<edm::InputTag>("eeSimHits"));
+    SimHits_inputfh_ = collector.consumes<edm::PCaloHitContainer>(conf.getParameter<edm::InputTag>("fhSimHits"));
+    is_Simhit_comp_ = conf.getParameter<bool>("isSimhitComp");
```

Comment:
The parameter can be retrieved in the constructor. And you probably don't need to consume the simhits if is_simhit_comp is false.